src/Components/SlateEditor/SlateEditor.js

```diff
@@ -124,6 +166,70 @@ export default function SlateEditor() {
     },
   };
 
+  const getCoordinates = () => {
+    let x, y;
+    const selection = window.getSelection();
+    if (selection.rangeCount !== 0) {
+      const range = selection.getRangeAt(0).cloneRange();
+      range.collapse(false);
+      const rect = range.getClientRects()[0];
+      if (rect) {
+        x = rect.top;
+        y = rect.left;
+      }
+    }
+    return { top: x > 120 ? x - 150 : x + 20, left: y };
+  };
+
+  const closeMenu = useCallback(() => {
+    setShowMenu(false);
+    document.removeEventListener("click", closeMenu);
+  }, []);
+
+  const openMenu = useCallback(() => {
+    const position = getCoordinates();
+    setCoordinates(position);
+    setShowMenu(true);
+    document.addEventListener("click", closeMenu);
+  }, [closeMenu]);
+
+  const renderMenu = () => {
```

Comment:
renderMenu variable name isn't very helpful in understanding what the variable contains; possibly cos you're looking at it as a function, and I'm looking at it as just a variable (see other comment on this)